torch/csrc/jit/runtime/profiling_graph_executor_impl.h

```diff
@@ -4,6 +4,17 @@
 namespace torch {
 namespace jit {
 
+enum class FusionBehavior { STATIC, DYNAMIC };
+
+using FusionStrategy = std::vector<std::pair<FusionBehavior, size_t>>;
+// E.g. {(Static, 2), (Dynamic, 10)}
+// Fuse with static shapes twice then fallback to 10 dynamic fusions, then stop
```

Comment:
Why would we want multiple dynamic fusions? What is the use-case for that?

Reply:
+1

Reply:
Even with "Dynamic" fusions, we are still specializing on rank, strides, dtype, device, etc. So if you think about a globally defined activation, there might end up with a number of dynamic specializations